### DISCUSSION TITLE
TemplateVariable: Provide Option "None" in Dashboard Dropdown, even if metrics with this labels exists

### DISCUSSION THREAD

**Discussion by bjung82** [grafana/grafana#38164] [Dashboards, Panels & Visualizations]
We have a list of four template variables, depending on each other - this works great :-)

1) Product
2) Customer Solution
3) Installation
4) Instance

There are some metrics in prometheus, that belong to a Product which, have an installation and an instance,
but have no customer solution at all (these are generic, for many customers).

The problem arises if there are metrics for the same product, but some of them have NO customer solution,
and some of them have. In this case, the dropdown for customer solution contains the label values of 
customer solutions, but **only** these. There is **no** Option "None" at the top of the dropdown **anymore**.

If there are only metrics for a product, all containing a label customer solution, this behaviour is ok.
If there are only metrics for a product, none containing a label customer solution, this behaviour is ok (None in dropdown).

**The problem is the mixed case: some have the label customer solution, some of them not. In this case, the
metrics without this label cannot be selected anymore.**

It would be nice to have an option on the variable: display "None" value.
Really nice would be an additional option to select this "None" value by default :-)

Are we missing something?

Now we are using a workaround by specifying a dummy label value in our prometheus metrics in this cases:
00_NA (this way it is displayed at the top of the drowdown list, and also selected automatically).
Automatic selection is nice to have, the other issue (no "None" value in dropdown in mixed case) seems
like a bug for me.

**Environment**:
- Grafana version: Grafana 7.2.0
- Data source type & version: Prometheus
- OS Grafana is installed on: Windows / Linux
- User OS & Browser: Linux / Windows + Chrome
- Grafana plugins: no plugins
- Others:

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #83008. Feel free to continue the discussion around this there. Thank you!
